lib/src/main/java/org/team100/lib/barcode/Sensor.java

```diff
@@ -0,0 +1,256 @@
+package org.team100.lib.barcode;
+
+import java.util.OptionalInt;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.DigitalOutput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * Represents the id sensor assembly, which consists of two parts:
+ * 
+ * * Pololu QTRX-HD-05A, a 5-channel reflective infrared photosensor
+ * * Sparkfun 16-channel multiplexer
+ * 
+ * The sensor provides 5 outputs, which are multiplexed to one RoboRIO analog
+ * input.
+ * 
+ * The multiplexer is controlled with 4 inputs, which are driven by RoboRIO
+ * digital outputs. The same multiplexer (and analog input) may be used for
+ * other inputs, e.g. a second photosensor, dip switches, etc.
+ * 
+ * The sensor emitters leak light to their neighbors, so there are two control
+ * pins which activate the "odd" and "even" emitters. These are connected to DIO
+ * outputs, and can be left "off" if the value isn't being read.
+ * 
+ * Note that the array uses *one based* numbering for the sensors and for the
+ * meaning of "odd" and "even," but the sensor pins themselves are wired in a
+ * zero-based way.
+ * 
+ * The LED controller appears to be an On Semi FAN5622 series, which responds
+ * to the CTRL signal in about 85 us to turn on, and 600us to turn off,
+ * which is fast but not that fast; we should read all the odds, flip the
+ * emitters,
+ * and then read all the evens.
+ * 
+ * https://www.onsemi.com/pdf/datasheet/fan5626-d.pdf
+ * 
+ * The photosensor on/off threshold also varies by position to account for light
+ * leakage.
+ * 
+ * The sensor used seems to be the Omron EE-SY1201 Reflective Photomicrosensor.
+ * 
+ * The output is a resistor pulling up and a photocell pulling down, so the
+ * output is inversely proportional to measures incident light: 0v is a lot,
+ * 5v is dark. so ambient light can drive it low and if the surface is too
+ * close, then the illuminator is shaded from the sensor, so a "light" surface
+ * can seem "dark" so you need the surface to be a few mm away -- actually the
+ * focal length is 3mm, so that's where the surface should be too.
+ * 
+ * Note that the response time of the sensor depends on the pull-up resistor
+ * value, and the Pololu board uses quite a large resistor, 47k ohms, which
+ * means that
+ * the rise and fall times are about 300 us.
+ * 
+ * Datasheet:
+ * 
+ * https://components.omron.com/us-en/datasheet_pdf/E602-E1.pdf
+ * 
+ * 
+ * Pololu product page:
+ * 
+ * https://www.pololu.com/product/4405/resources
+ */
+public class Sensor {
+    private static final int loops = 150;
+
+    private final double[] s5 = new double[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```

Comment:
i think you can replace this with new double[150], unless you want to specify something other than all zeros?